source/common/buffer/buffer_impl.h

```diff
@@ -626,7 +626,7 @@ class BufferFragmentImpl : NonCopyable, public BufferFragment {
   const std::function<void(const void*, size_t, const BufferFragmentImpl*)> releasor_;
 };
 
-class LibEventInstance : public Instance {
+class LibEventInstance : public Instance, public Chunker {
```

Comment:
We shouldn't need to change BufferImpl at all, or add the complexity of multiple inheritance here.

Reply:
I met an issue:
```
Http::Code AdminImpl::request(absl::string_view path_and_query, absl::string_view method,
                              Http::ResponseHeaderMap& response_headers, std::string& body) {
  AdminFilter filter(createCallbackFunction());

  auto request_headers = Http::RequestHeaderMapImpl::create();
  request_headers->setMethod(method);
  filter.decodeHeaders(*request_headers, false);
  Buffer::OwnedImpl response;   <===

  Http::Code code = runCallback(path_and_query, response_headers, response, filter);   <===
  Utility::populateFallbackResponseHeaders(code, response_headers);
  body = response.toString();
  return code;
}
```

`Chunker` can not use anything in `Buffer::OwnedImpl`, I do not find a good way to do it :(

Reply:
I think Chunker could contain a Buffer::OwnedImpl (not inherit from it), and delegate to it. draining as needed to keep it under some max size. It will need to be passed, in its constructor, a network socket it can use to drain bytes as it hits its buffering limit.

If this is seeming too complex at this point we can assign someone else to do it. The work you've done so far is actually pretty helpful and you could finish that off by just (for now) making "Chunker" an alias for buffer in admin.h:
```
using Chunker == Buffer::Instance;
```
and then in a further PR we can actually make a real implementation of Chunker that auto-drains into the network.

Reply:
Got it, thanks for your patience and help, I learn a lot from you @jmarantz !